qcengine/programs/tests/test_nwchem.py

```diff
@@ -43,3 +44,45 @@ def test_b3lyp(nh2):
     assert res["properties"]["calcinfo_natom"] == 3
     assert res["properties"]["calcinfo_nalpha"] == 5
     assert res["properties"]["calcinfo_nbasis"] == 13
+
+
+@using("nwchem")
+def test_hess(nh2):
+    resi = {"molecule": nh2, "driver": "hessian", "model": {"method": "b3lyp", "basis": "3-21g"}}
+    res = qcng.compute(resi, "nwchem", raise_error=True, return_dict=True)
+    assert compare_values(-3.5980754370e-02, res["return_result"][0], atol=1e-3)
```

Comment:
Unsure if this should work a `return_result` of a Hessian computation should be a 2-d array. I would also try to test more values than just a diagonal element to check the rotation as well.

Reply:
I'm unsure, based on the `Array` type for `AtomicResult.return_result`. as well. I'll switch it to a 2D array, as that makes more sense to me.

Reply:
Ok, took me a bit to figure this out, but the base class for AtomicResult (ProtoModel) flattens all arrays when JSON serializing them. So, return_result will also be flattened regardless of what I do if I return a dictionary from `qcng.compute`

Reply:
Ah, hmm. This must date back to some legacy code:
```python
    if return_dict:
        return json.loads(ret.json())
    else:
        return ret
```

We probably need to make an issue out of this where `return_dict` will return NumPy arrays which I think makes a bit more sense than the current formulation.

Reply:
I do think we need to test a few more elements here to prevent the exact issue that you hit of triu/tril.

Reply:
Alright, the test now uses a non-dict return type - a 2D numpy array